### PR TITLE
Add tests for exception classes

### PR DESCRIPTION
**Pull request by malukenho**

### CODE REVIEW

File: test/Exception/BadMethodCallExceptionTest.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+use Zend\Expressive\Exception\ExceptionInterface;
+
+final class BadMethodCallExceptionTest extends \PHPUnit_Framework_TestCase
+{
+    public function testException()
+    {
+        $exception = new BadMethodCallException();
+
+        $this->assertInstanceOf(ExceptionInterface::class, $exception);
+        $this->assertInstanceOf(BadMethodCallException::class, $exception);
```
Comment (danizord) [inline test/Exception/BadMethodCallExceptionTest.php]: This one is useless given the behavior of `new` in PHP? :)

File: test/Exception/BadMethodCallExceptionTest.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+{
+    public function testException()
+    {
+        $exception = new BadMethodCallException();
+
+        $this->assertInstanceOf(ExceptionInterface::class, $exception);
+
+        $this->setExpectedException(BadMethodCallException::class);
+
+        throw $exception;
```
Comment (Ocramius) [inline test/Exception/BadMethodCallExceptionTest.php]: As discussed before, this is not the right way to test an exception in my opinion, but I'll have to think about it.

**Comment by weierophinney** [comment]
In the majority of cases, these are simply _types_ that are thrown and/or caught internally, and are thus covered elsewhere. I'd argue that if we need coverage for specific exceptions, we would want to write tests that ensure that code that would raise or catch one of these types is exercised, and then annotate that the test covers that exception type.

**Comment by weierophinney** [comment]
BTW, #414 now covers several of these exceptions via new unit tests.

**Comment by malukenho** [comment]
Closing as not necessary

**Pull request closed without merging**